crates/driver/src/infra/blockchain/contracts.rs

```diff
@@ -1 +1,49 @@
+use {crate::domain::eth, ethcontract::Web3};
+
 pub use crate::boundary::contracts::{GPv2Settlement, ERC20, WETH9};
+
+#[derive(Debug, Clone)]
+pub struct Contracts {
+    settlement: contracts::GPv2Settlement,
+    weth: contracts::WETH9,
+}
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Addresses {
+    pub settlement: Option<eth::ContractAddress>,
+    pub weth: Option<eth::ContractAddress>,
+}
```

Comment:
Would be cool if we only ever had to initialize those addresses once. In that case we could make `settlement` and `weth` required because the service will not be useful when those addresses are unknown.
Just some idea.

Reply:
Ahh, I misunderstood the code. It's a struct to configure `Contracts` in case you want to set some specific values. 👌